# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1229630780 -3600
#      Thu Dec 18 21:06:20 2008 +0100
# Node ID b8de157b4948756598844e2518340737d817532c
# Parent  970b4dbff9e470b49897a4f3791dc8142823d123
ChangeLog entry and style fixes for last two patches

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,14 @@
+2008-12-18  Daniel J Sebald <daniel.sebald@ieee.org>
+
+	* time/datevec.m (__date_vfmt2sfmt__): New helper function.
+	(datevec): Avoid repeated parsing of the format string.
+	* set/unique.m: Only check for options if nargin > 1.
+
 2008-12-15  Jaroslav Hajek  <highegg@gmail.com>
 
 	* optimization/lsqnonneg.m: Preprocess using QR for over-determined
 	systems. Simplify & fix indexing. Use left division for step problem.
 	Fix output args.
 
 2008-12-13  Francesco Potortì  <pot@gnu.org>
 
diff --git a/scripts/set/unique.m b/scripts/set/unique.m
--- a/scripts/set/unique.m
+++ b/scripts/set/unique.m
@@ -44,17 +44,17 @@ function [y, i, j] = unique (x, varargin
   if (nargin < 1)
     print_usage ();
   endif
 
   if (nargin > 1)
 
     ## parse options
     if (iscellstr (varargin))
-      varargin = unique(varargin);
+      varargin = unique (varargin);
       optfirst = strmatch ('first', varargin) > 0;
       optlast = strmatch ('last', varargin) > 0;
       optrows = strmatch ('rows', varargin) > 0 && size (x, 2) > 1;
       if (optfirst && optlast)
         error ("unique: cannot specify both 'last' and 'first'.");
       elseif (optfirst + optlast + optrows != nargin-1)
         error ("unique: invalid option.");
       endif
@@ -103,17 +103,17 @@ function [y, i, j] = unique (x, varargin
       match = (y(1:n-1) == y(2:n));
     endif
     idx = find (match);
     y(idx) = [];
   endif
 
   if (nargout >= 3)
     j = i;
-    j(i) = cumsum ([1 !match]);
+    j(i) = cumsum ([1, !match]);
   endif
 
   if (optfirst)
     i(idx+1) = [];
   else
     i(idx) = [];
   endif
 
